include/envoy/secret/secret_manager.h

```diff
@@ -17,14 +21,37 @@ class SecretManager {
   virtual ~SecretManager() {}
 
   /**
+   * @param secret a raw pointer of the derived class of Secret.
+   */
+  template <typename T> void addOrUpdateSecret(T* secret) {
```

Comment:
It is so clear about the owership transfer when using raw pointer.  Can we pass in shared_ptr<T> ?

Reply:
Changed variable type